frontend/src/scenes/surveys/surveyLogic.tsx

```diff
@@ -147,7 +148,42 @@ export interface SurveyDateRange {
     date_to: string | null
 }
 
-const getResponseField = (i: number): string => (i === 0 ? '$survey_response' : `$survey_response_${i}`)
+// New function that supports both index-based and ID-based approaches
```

Comment:
Are all methods here used elsewhere? otherwise we should not default to `export`

Reply:
need export to do tests

Reply:
no hard copying isn't ideal.
I usually would test the 1st method to be called and just assert the final result; your method input has to allow running into all code paths with the inner methods as well of course, it's more like an integration method that test multiple methods rather than every single method, I usually find this approach better, and a must on SDKs since you cant exposeeverything otherwise everything is a public API, here is different but I think it still makes sense if easy to apply the same concept